src/services/catalog.service.ts

```diff
@@ -0,0 +1,152 @@
+import * as _ from 'lodash';
+
+export class CatalogService {
+  static $inject = ['Constants', '$filter'];
+
+  public $filter: any;
+  private constants: any;
+  private categories: any;
+
+  constructor(constants: any, $filter: any, $scope: any) {
+    this.constants = constants;
+    this.$filter = $filter;
+    this.categories = this.constants.SERVICE_CATALOG_CATEGORIES;
+  }
+
+  public getServiceItem(resource: any) {
+    return new ServiceItem(resource, this);
+  }
+
+  public getImageItem(resource: any) {
+    return new ImageItem(resource, this);
+  }
+
+  public getCategoriesBySubCategories(tags: any) {
+    let catsBySubCats = {};
+    _.each(tags, (tag) => {
+      _.each(this.categories, (category) => {
+        let subCat: any = _.find(category.subCategories, (subCategory: any) => {
```

Comment:
This may come back to my initial comment: "I still need to reconcile [catalog categories](https://github.com/openshift/origin-web-catalog/blob/master/src/constants.ts#L12) with [consoles categories](https://github.com/openshift/origin-web-console/blob/master/app/scripts/constants.js#L301).  Need to deep-dive with someone and determine what else needs to be added to the Catalog's categories."

I see in  [consoles categories](https://github.com/openshift/origin-web-console/blob/master/app/scripts/constants.js#L301) there is something called 'subcategories', which really makes that structure 3 levels deep:  Categories -> Items -> subcategories.  I am currently using a categories constant which is two levels deep.